Documentation updates 06 2022

tonhuisman opened the pull request:
Updates for some documentation pages, adjusted where needed, either by updated/changed software or outdated information.

- Add a missing screenshot for Tools/Advanced (Rules options)
- Update extension list for VSCode
- Remove Atom as a development environment, as it is no longer maintained
- Improve styling (contrast), 'changed', add RETIRED plugin category
- Update documentation page on writing documentation  (non-recursive 😛)
- Procedure for configuring a RTC chip

Resolves #4122

tonhuisman committed:
[Docs] Add missing Tools/Advanced (Rules) screenshot

tonhuisman committed:
[Docs] Update extensions list for VSCode, remove Atom reference, update some text

tonhuisman committed:
[Docs] Fix some heading level warnings

TD-er commented on docs/source/Participate/Documentation.rst:
```diff
@@ -12,17 +12,25 @@ for more documentation which has not been moved here.
+As an alternative, VSCode can be used as a development environment. See also :ref:`PlatformIO_page`
 
-With the ESPEasy project open in Atom, open the PIO terminal in Atom.
+With the ESPEasy project open in VSCode, open the PIO terminal in VSCode.
```
PIO terminal?
Maybe better to show a screenshot like this:

![image](https://user-images.githubusercontent.com/3751318/174387929-59c1ecb6-ec1d-485a-a2bb-69b4cf85e362.png)

tonhuisman committed:
[Docs] Update styling contrast, add gray and changed options, Retired plugin category, apply changed where appropriate

tonhuisman committed:
[Docs] Add improved reference to open PIO Terminal

tonhuisman commented:
How it currently is:
![Screenshot - 17-06-2022 , 21_40_46](https://user-images.githubusercontent.com/1922393/174391329-1b8447ee-ee14-48eb-aef0-b7933565d60c.png)

With improved contrast it looks like this:
![Screenshot - 17-06-2022 , 21_36_01](https://user-images.githubusercontent.com/1922393/174390891-dd9032f7-d549-4457-a056-2511ab1f22c5.png)

Changed similarly for red: (and restored 'Development', even though no plugin currently has that tag)
![Screenshot - 17-06-2022 , 21_39_34](https://user-images.githubusercontent.com/1922393/174391165-42eb6b22-4168-4604-9b2e-3bddb8b4bf9d.png)

And Changed: (P019)
![Screenshot - 17-06-2022 , 21_43_32](https://user-images.githubusercontent.com/1922393/174391597-b1fee12c-ee73-4114-9d0e-17e499b80623.png)

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/some-documentation-updates-06-2022

tonhuisman committed:
[Docs] Updates on writing documentation, remove unused Linux pdf generation setup

tonhuisman committed:
Merge branch 'mega' into feature/some-documentation-updates-06-2022

tonhuisman committed:
[Docs] Procedure for configuring a RTC chip

tonhuisman committed:
[Docs] Add alternatives for RTC configuration, update Rules options after recent change

TD-er committed:
Merge branch 'mega' into feature/some-documentation-updates-06-2022
